### PR TITLE
Add contents for account dref forms

### PR DESCRIPTION
**Pull request by frozenhelium**

### CODE REVIEW

File: src/utils/common.ts [hunk 1]
```diff
@@ -405,3 +405,39 @@ export function denormalizeList<ListItem, SecondaryListItem, ReturnType>(
...
+            return false;
+        }
+
+        return item.some(hasSomeDefinedValue);
+    }
+
+    if (typeof item === 'object') {
+        if (!item) {
+            return false;
+        }
```
Comment (tnagorra) [inline src/utils/common.ts:437]: Do we even need this check?

File: src/views/AccountDrefApplications/index.tsx [hunk 2]
```diff
@@ -1,12 +1,45 @@
...
+            {currentView === 'active' && (
+                <ActiveDrefTable
+                    actions={(
+                        <Button
+                            name="completed"
+                            onClick={setCurrentView}
+                            variant="tertiary"
+                            actions={<ChevronRightLineIcon className={styles.icon} />}
+                        >
+                            Show completed DREFs
```
Comment (tnagorra) [inline src/views/AccountDrefApplications/index.tsx]: Add FIXME note for translation

File: src/views/AccountDrefApplications/index.tsx [hunk 3]
```diff
@@ -1,12 +1,45 @@
...
+            {currentView === 'completed' && (
+                <CompletedDrefTable
+                    actions={(
+                        <Button
+                            name="active"
+                            onClick={setCurrentView}
+                            variant="tertiary"
+                            icons={<ChevronLeftLineIcon className={styles.icon} />}
+                        >
+                            Back to active DREFs
```
Comment (tnagorra) [inline src/views/AccountDrefApplications/index.tsx]: Add FIXME note for translation

File: src/views/AccountDrefApplications/DrefTableActions/index.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+import Link from '#components/Link';
+import useTranslation from '#hooks/useTranslation';
+import RouteContext from '#contexts/route';
+import { components } from '#generated/types';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+const DREF_STATUS_COMPLETED = 1;
+const DREF_STATUS_IN_PROGRESS = 0;
```
Comment (tnagorra) [inline src/views/AccountDrefApplications/DrefTableActions/index.tsx]: Let's move this to utils/constants and also use `satisfies` that the enum value is valid.

File: src/views/AccountDrefApplications/DrefTableActions/index.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    const editPath = useMemo(
+        () => {
+            if (applicationType === 'DREF') {
+                return generatePath(drefApplicationFormRoute.absolutePath, { drefId: id });
+            }
+
+            return undefined;
+        },
+        [id, applicationType, drefApplicationFormRoute],
+    );
```
Comment (tnagorra) [inline src/views/AccountDrefApplications/DrefTableActions/index.tsx]: Why not just write this directly on render?

File: src/views/AccountDrefApplications/ActiveDrefTable/index.tsx [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+            const {
+                final_report_details,
+                operational_update_details,
+                has_ops_update,
+                has_final_report,
+            } = dref;
+
+            if (has_final_report) {
+                type FinalReportType = typeof final_report_details;
+                const finalReportList = final_report_details as unknown as FinalReportType[];
```
Comment (tnagorra) [inline src/views/AccountDrefApplications/ActiveDrefTable/index.tsx:89]: Why are we casting this array
Let's add NOTE

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/utils/common.ts (hunk 1 above)
Comment (frozenhelium) [reply src/utils/common.ts:437]: typescript will cast it as `object | null`

File: src/views/AccountDrefApplications/DrefTableActions/index.tsx (hunk 5 above)
Comment (frozenhelium) [reply src/views/AccountDrefApplications/DrefTableActions/index.tsx]: We'll have 2 more type of links for the forms (Ops Update and Final Report) I'll add TODOs

File: src/views/AccountDrefApplications/ActiveDrefTable/index.tsx (hunk 6 above)
Comment (frozenhelium) [reply src/views/AccountDrefApplications/ActiveDrefTable/index.tsx:89]: We have notes slightly above

Comment (tnagorra) [reply src/views/AccountDrefApplications/ActiveDrefTable/index.tsx:89]: Let's add NOTE anywhere we are casting data. It's easier and we don't miss it.

File: src/utils/common.ts (hunk 1 above)
Comment (tnagorra) [reply src/utils/common.ts:437]: We have already used isDefined on the top. So, it shouldn't be the case.

File: src/utils/constants.ts [hunk 7]
```diff
@@ -5,6 +5,7 @@ import {
 type OperationTypeEnum = components['schemas']['OperationTypeEnum'];
 type StatusTypeEnum = components['schemas']['Key1d2Enum'];
 type ProgrammeTypeEnum = components['schemas']['Key1d2Enum'];
+type DrefStatus = components['schemas']['OperationTypeEnum'];
```
Comment (tnagorra) [inline src/utils/constants.ts:9]: This is duplicated.

Comment (tnagorra) [reply src/utils/constants.ts:9]: Is this a mistake.

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/utils/constants.ts (hunk 7 above)
Comment (frozenhelium) [reply src/utils/constants.ts:9]: Might need to create a separate enum in the server

**Review by tnagorra — APPROVED** [review]

**Pull request merged**